htan_registry/tiffinfo.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import os
+import subprocess
+import re
+
+from synapsegenie.example_filetype_format import FileTypeFormat
+from synapsegenie import process_functions
+
+logger = logging.getLogger(__name__)
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+
+class TiffInfo(FileTypeFormat):
+
+    _filetype = "tiffinfo"
+
+    _process_kwargs = ["databaseSynId"]
+
+    def _validate_filetype(self, filePath):
+        assert os.path.basename(filePath[0]).endswith(".tif{1,2}|.svs")
+
+    def _process(self, df):
+        df.columns = [df.upper() for col in df.columns]
+        return df
+
+    def read_file(self, filePathList):
+        '''
+        Each file is to be read in for validation and processing.
+        This is not to be changed in any functions.
+
+        Args:
+            filePathList:  A list of file paths (Max is 2 for the two
+                           clinical files)
+
+        Returns:
+            df: Pandas dataframe of file
+        '''
+        # This file isn't a dataframe, so just return the filepath
+        filepath = filePathList[0]
+        return filepath
+
+    def process_steps(self, df, newPath, databaseSynId):
+        df = self._process(df)
+        process_functions.updateData(self.syn, databaseSynId, df, self.center,
+                                     toDelete=True)
+        df.to_csv(newPath, sep="\t", index=False)
+        return newPath
+
+    def _validate(self, path_or_df):
+        total_error = []
+        warning = []
+
+        # Validate the OME-XML with xmlvalid
+        xmlvalid_cmd = os.path.join(here, '../bftools/xmlvalid')
```

Comment:
Whats the difference between this and the ometif class?

Reply:
`ometif` class to explicitly validate `ome-tiff` files. `tiffinfo` class intended to validate and basic PHI check any tiff based file. Can have as one class if better.

Reply:
Hmmm. Let's discuss more in our chat.  Basically what we can have are more modular validation functions in a separate module and import them.  Then you have different file format classes like `class Svs()` and `class OmeTiff()` and `class Tiff()` that can call the same functions, BUT if you wanted to add extra validation specific to each of those file formats, you then could.